docs/hosting/deploy.md

```diff
@@ -1,20 +1,417 @@
+
+Reflex’s hosting service makes it easy to deploy your apps without worrying about configuring the infrastructure.
+
+```md alert info
+This tutorial assumes you have successfully `reflex init` and `reflex run` your app.
+```
+
+### Authentication
+
+```md alert info
+Hosting service requires `reflex>=0.3.2`.
+```
+
+First, create an account or log into it using the following command.
+
+```bash
+reflex login
+```
+
+You will be redirected to your browser where you can authenticate through Github or Gmail.
+
+### Deployment
+
+Once you have successfully authenticated, you can start deploying your apps.
+
+```md alert warning
+Make sure you have a `requirements.txt`  file at the top level app directory that contains all your python dependencies!
+```
+
+Navigate to the project directory that you want to deploy and type the following command:
+
+```bash
+reflex deploy
+```
+
+The command is by default interactive. It asks you a few questions for information required for the deployment.
+
+**Name**: choose a name for the deployed app. This name will be part of the deployed app URL, i.e. `<app-name>.reflex.run`. The name should only contain domain name safe characters: no slashes, no underscores. Domain names are case insensitive. To avoid confusion, the name you choose here is also case insensitive. If you enter letters in upper cases, we automatically convert them to lower cases.
+
+**Regions**: enter the region code here or press `Enter` to accept the default. The default code `sjc` stands for San Jose, California in the US west coast. Check the list of supported regions at [reflex deployments regions](#reflex-deployments-regions).
+
+**Envs**: `Envs` are environment variables. You might not have used them at all in your app. In that case, press `Enter` to skip. More on the environment variables in the later section [Environment Variables](#environment-variables).
+
+That’s it! You should receive some feedback on the progress of your deployment and in a few minutes your app should be up. 🎉
+
+```md alert info
+Once your code is uploaded, the hosting service will start the deployment. After a complete upload, exiting from the command **does not** affect the deployment process. The command prints a message when you can safely close it without affecting the deployment.
+```
+
+## Concepts
+
+### Requirements
+
+To be able to deploy your app, we ask that you prepare a `requirements.txt` file containing all the required Python packages for it. The hosting service runs a `pip install` command based on this file to prepare the instances that run your app. We recommend that  you use a Python virtual environment when starting a new app, and only install the necessary packages. This reduces the preparation time installing no more packages than needed, and your app is deployed faster. There are a lot of resources online on Python virtual environment tools and how to capture the packages in a `requirements.txt` file.
+
+### Environment Variables
```

Comment:
might be nice here just to show a screenshot of what the command looks like and then an example of what an API key entry might look like (for a fake key of course)

Reply:
there's an example in the deploy command. we'll also have a video tutorial of all the steps of a deployment